ngraph/frontend/onnx_import/src/op/size.cpp

```diff
@@ -34,13 +34,9 @@ namespace ngraph
                 OutputVector size(const Node& node)
                 {
                     auto data = node.get_ng_inputs().at(0);
-                    std::int64_t tensor_elements_count{
-                        static_cast<std::int64_t>(shape_size(data.get_shape()))};
-
-                    return {std::make_shared<default_opset::Constant>(
-                        ngraph::element::i64,
-                        Shape{},
-                        std::vector<std::int64_t>{tensor_elements_count})};
+                    auto axes = default_opset::Constant::create(ngraph::element::i32, Shape{}, {0});
+                    auto input_shape = std::make_shared<default_opset::ShapeOf>(data);
+                    return {std::make_shared<default_opset::ReduceProd>(input_shape, axes)};
```

Comment:
Why do you return a ReduceProd(ShapeOf) instead of the ShapeOf itself?

Reply:
According to the ONNX spec Size op should return "scalar that equals to the total number of elements of the input tensor". So ShapeOf is not enough, it returns a vector with shape dimensions. 
We need ReuceProd to get scalar value.

For example, Node has `Shape{1, 2, 4, 1, 3}`, ShapeOf(Node) will return `vector {1, 2, 4, 1, 3}`, ReudceProd(ShapeOf, axis=0) will return multiplication `1*2*4*1*3` so `scalar` with value 24 and this is exactly what we need.

Reply:
Right, my bad, I was looking at a different op in the docs...